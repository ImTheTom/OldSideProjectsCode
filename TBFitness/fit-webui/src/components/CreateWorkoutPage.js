import React from "react";
import Header from "./Header";
import axios from "axios";
import Config from "../config";
import { Input, Button } from "antd";

class CreateWorkoutPage extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			Name: "",
		};
		this.handleInputChange = this.handleInputChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleInputChange(event) {
		const target = event.target;
		const value = target.value;
		const name = target.name;

		this.setState({
			[name]: value
		});
	}

	handleSubmit(event) {
		event.preventDefault();
		axios.post(Config.host.NAME + "workout", {
			id: 0,
			user_id: 1,
			name: this.state.Name,
		}).then(function (response) {
			console.log(response);
		}).catch(function (error) {
			console.log(error);
		});
	}

	render() {
		return (
			<div>
				<div>
					<Header />
				</div>
				<div className="CreateBox">
					<div className="create-lift-title">
		Create Workout
					</div>
					<form onSubmit={this.handleSubmit}>
						<Input className="create-lift-input" placeholder="Name" label="Name" name="Name" value={this.state.Name} onChange={(e) => this.handleInputChange(e)} /> <br></br>
						<Button className="create-lift-input" type="primary">Create Workout</Button>
					</form>
				</div>
			</div>
		);
	}
}


export default CreateWorkoutPage;

