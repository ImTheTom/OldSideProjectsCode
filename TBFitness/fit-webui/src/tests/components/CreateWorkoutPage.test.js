import CreateWorkoutPage from "../../components/CreateWorkoutPage";
import React from "react";
import { shallow } from "enzyme";

test("should render create Lift page correctly", () => {
	const wrapper = shallow(<CreateWorkoutPage />);
	expect(wrapper).toMatchSnapshot();
});