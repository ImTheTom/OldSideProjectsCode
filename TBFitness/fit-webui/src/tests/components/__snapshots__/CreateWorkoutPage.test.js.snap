// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render create Lift page correctly 1`] = `
<div>
  <div>
    <Header />
  </div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label>
        Name
        <input
          name="Name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <button>
        Create
      </button>
    </form>
  </div>
</div>
`;
